### DISCUSSION TITLE
Query Regarding FlatMode Configuration in MTConnect CPP Agent

### DISCUSSION THREAD

**Discussion by Pranay265** [mtconnect/cppagent#432] [Q&A]
I am currently working with the MTConnect CPP Agent 2.3.0.3 and have a question regarding the FlatMode configuration option in the agent's agent.cfg file.

I noticed that when I enable or disable FlatMode (e.g., FlatMode = true or FlatMode = false), it doesn't seem to have any effect on the MQTT topic structure generated by the agent. Specifically, I am trying to achieve a flat topic structure like **MTConnect/Devices/{deviceuuid}/Observations/{DataItemID}.**

Here is an example of my sink configuration in the agent.cfg file:
Sinks {
  MqttService {
    Server = localhost
    Port = 1883
    ObservationTopic = MTConnect/Devices/
    FlatMode = true
  }
}

I expected that enabling FlatMode would flatten the MQTT topic structure, but I haven't observed any change in the topics generated.

Could you please provide some insights into how FlatMode should be correctly configured to achieve a flat topic structure in the MQTT output of the MTConnect CPP Agent? Additionally, are there any specific considerations or dependencies that I should be aware of when using FlatMode?

@wsobel : Any guidance or clarifications on this matter would be greatly appreciated.

**Comment by wsobel** [comment]
I don't see any documentation for FlatMode or in the configuration of the MQTT Service. I have checked and not sure where the docs are coming from. We may had deprecated since it was not working.

I am deprecating the old version of the MQTT implementation so flat mode should not be an issue anymore. 

The Mqtt2Service (In 2.5 it will be the MqttService). This provides a simple set of three documents: Probe (Devices), Current (snapshots every 10 seconds) and (Samples every second RBE). 

See if that works better. I would also like to understand what you are trying to do so I can better address your question.
